resources/site.rb

```diff
@@ -18,20 +17,211 @@
+  # 'SITE "Default Web Site" (id:1,bindings:http/*:80:,state:Started)'
+  Chef::Log.debug("#{desired} list site command output: #{cmd.stdout}")
+  if cmd.stderr.empty?
+    result = cmd.stdout.gsub(/\r\n?/, "\n") # ensure we have no carriage returns
+    result = result.match(/^SITE\s\"(?<site>#{desired.site_name})\"\s\(id:(?<site_id>.*),bindings:(?<bindings>.*),state:(?<state>.*)\)$/i)
+    Chef::Log.debug("#{desired} current_resource match output: #{result}")
+    if result
+      site_id result[:site_id].to_i
+      bindings result[:bindings]
+      running result[:state] =~ /Started/ ? true : false
+    else
+      running false
+    end
+
+    if site_id
+      values = "#{bindings},".match(%r{(?<protocol>[^\/]+)\/\*:(?<port>[^:]+):(?<host_header>[^,]*),})
+      # get current values
+      cmd = "#{appcmd(node)} list site \"#{site_name}\" /config:* /xml"
+      Chef::Log.debug(cmd)
+      cmd = shell_out cmd
+      if cmd.stderr.empty?
+        xml = cmd.stdout
+        doc = Document.new(xml)
+        path value doc.root, 'SITE/site/application/virtualDirectory/@physicalPath'
+        log_directory value doc.root, 'SITE/site/logFile/@directory'
+        log_period value(doc.root, 'SITE/site/logFile/@period').to_sym
+        log_truncsize value(doc.root, 'SITE/site/logFile/@truncateSize').to_i
+        application_pool value doc.root, 'SITE/site/application/@applicationPool'
+      end
+
+      if values
+        protocol values[:protocol].to_sym
+        port values[:port].to_i
+        host_header values[:host_header]
+      end
+    else
+      running false
+    end
+
+    if values
+      protocol values[:protocol]
+      port values[:port].to_i
+      host_header values[:host_header]
+    end
+  else
+    log "Failed to run iis_site action :config, #{cmd.stderr}" do
+      level :warn
+    end
+  end
+end
+
+action :add do
+  if !current_resource.site_id
+    converge_by "Created the Site - \"#{new_resource}\"" do
+      cmd = "#{appcmd(node)} add site /name:\"#{new_resource.site_name}\""
+      cmd << " /id:#{new_resource.site_id}" if new_resource.site_id
+      cmd << " /physicalPath:\"#{windows_cleanpath(new_resource.path)}\"" if new_resource.path
+      if new_resource.bindings
+        cmd << " /bindings:\"#{new_resource.bindings}\""
+      else
+        cmd << " /bindings:#{new_resource.protocol}/*"
+        cmd << ":#{new_resource.port}:" if new_resource.port
+        cmd << new_resource.host_header if new_resource.host_header
+      end
+
+      # support for additional options -logDir, -limits, -ftpServer, etc...
+      cmd << " #{new_resource.options}" if new_resource.options
+      shell_out!(cmd, returns: [0, 42])
+
+      configure
+
+      if new_resource.application_pool
+        shell_out!("#{appcmd(node)} set site /site.name:\"#{new_resource.site_name}\" /[path='/'].applicationPool:\"#{new_resource.application_pool}\"", returns: [0, 42])
+      end
+      new_resource.updated_by_last_action(true)
```

Comment:
we should never be using the updated_by_last_action with modern chef code, instead should be doing converge_by blocks.